src/XrdHttp/XrdHttpReq.cc

```diff
@@ -2369,7 +2370,11 @@ int XrdHttpReq::PostProcessHTTPReq(bool final_) {
 
 
         // If we are here it's too late to send a proper error message...
-        if (xrdresp == kXR_error) return -1;
```

Comment:
We can't do this, unfortunately.

Per the comment above, we have already started the response and HTTP rules are that you can't send a second response to the same request.  This change breaks the HTTP protocol framing.

For `GET`, we have a small extension where the client can opt-in to a trailer (which is part of the HTTP protocol though rarely used) that indicates whether an error occurred while sending the `GET` response.  We could do a similar thing for `PUT` -- however, for now, we should keep the behavior of just abruptly closing the socket and save the refactor of the `PUT` response for later.